prov/util/src/util_cq.c

```diff
@@ -451,6 +451,7 @@ static void util_peer_cq_cleanup(struct util_cq *cq)
 
 	util_comp_cirq_free(cq->cirq);
 	free(cq->src);
+	free(cq->peer_cq);
```

Comment:
Although functionally equivalent right now, would it be more semantically correct to use `fi_close(cq->peer_cq)` here?

Reply:
`fi_close(&cq->peer_cq->fid)`

Reply:
This path matches up with the code path that calls calloc().  I agree calling fi_close() seems desirable, but then I'd include other changes.  For example, most or all of the cleanup inside this function (util_peer_cq_cleanup) should be handled in fi_close().  The separation between the util_cq and peer_cq isn't clean and could be better.

Reply:
util_init_peer_cq() allocates and initializes more than the 'peer_cq'.  util_peer_cq_cleanup() aligns with util_init_peer_cq() for cleanup purposes.  All of these fields in util_cq are basically the peer cq.
```
	struct fid_peer_cq	*peer_cq;

	/* Only valid if not FI_PEER */
	struct util_comp_cirq	*cirq;
	fi_addr_t		*src;
	struct slist		aux_queue;
	fi_cq_read_func		read_entry;
```
We could wrap those into another struct.  I think we skipped doing that because of the code churn, decreased code readability, and would only save 4 pointers of memory per CQ in the peer use case.

Reply:
I don't think those cleanups in `util_peer_cq_cleanup` need to be handled inside `fi_close()` because what is being closed is `util_cq->peer_cq`, not `util_cq` itself. The close function of `util_cq` itself does call the `util_cq_cleanup` which in turn calls `util_peer_cq_cleanup` so the flow is correct. On the other hand, Closing `util_cq->peer_cq` only involves freeing the allocation, which is exactly what is done inside `util_peer_cq_close()`, which is already assigned as the close method of the corresponding `fid_ops`.